src/parsers/installrdf.js

```diff
@@ -0,0 +1,56 @@
+import { ADDON_TYPE_MAP, INSTALL_RDF } from 'const';
+import { TYPE_INVALID,
+         TYPE_MISSING } from 'messages';
+
+import log from 'logger';
+import RDFScanner from 'scanners/rdf';
+
+
+export default class InstallRdfParser {
+
+  constructor(rdfString, collector) {
+    this.rdfString = rdfString;
+    // Provides ability to directly add messages to
+    // the collector.
+    this.collector = collector;
```

Comment:
I thought of parsers as simply parsing data for the scanners to look through, and with the exception of things like parse errors they wouldn't handle errors. Things like the `TYPE_INVALID` and `TYPE_MISSING` errors seem like they'd go in the scanner, while the parser is just the code that converts a format (in this case RDF) into a JS API/object for us to look through.